Apps/VisionOS/VisionOS/Storage/AppState.swift

```diff
@@ -1,5 +1,25 @@
 import Foundation
 
+extension [InlineNavigationLink]: UserDefaultsCodable {
+    static func storageKey() -> String {
+        "navigationPath"
+    }
+
+    static func empty() -> [Element] {
+        []
+    }
+}
+
+extension Set<InlineNavigationLink>: UserDefaultsCodable {
+    static func storageKey() -> String {
+        "visitedLinks"
+    }
+
+    static func empty() -> Set<Element> {
+        .init()
+    }
+}
```

Comment:
Why are we saving this information to persistent storage?

Reply:
This is a persistent UI state. Helps the user identify screens they already visited before.
We have few things to store
Persistent
- Data used for the SDK like the workspace settings and initialize (profile) details
- Navigation state so the user don't always start on the landing screen (a bit of a UI convenience)

Non-Persistent
These are moved to a ViewModel that won't be persistent
- Current screen title - updated on each screen onAppear
- error and success message

Now we can argue that navigation is part of the view model. Which isn't wrong. It would just mean:
- Persistent storage used for SDK related payloads
- Persistent storage for parts of the view model
- Non-persistent storage for other parts of the view model

Again, nothing wrong with that, just didn't encounter a reason to see the full value of adding this strict separation yet in the app